alibi_detect/cd/utils.py

```diff
@@ -39,3 +43,67 @@ def update_reference(X_ref: np.ndarray,
             raise KeyError('Only `reservoir_sampling` and `last` are valid update options for X_ref.')
     else:
         return X_ref
+
+
+def activate_train_mode_for_dropout_layers(model: Union[nn.Module, tf.keras.Model], backend: str) -> Callable:
+    # TODO: Figure a way to do this properly for tensorflow models.
+    if backend == 'pytorch':
+        model.eval()
+        n_dropout_layers = 0
+        for module in model.modules():
+            if isinstance(module, nn.Dropout):
+                module.train()
+                n_dropout_layers += 1
+        if n_dropout_layers == 0:
+            raise ValueError("No dropout layers identified.")
+    elif backend == 'tensorflow':
+        model.trainable = False
+        model = partial(model, training=True)  # Note this affects batchnorm etc also
+    else:
+        raise NotImplementedError("Only 'pytorch' or 'tensorflow' backends supported")
+
+    return model
+
+
+def get_preds(
+    x: np.ndarray,
+    model: Callable,
+    backend: str,
+    batch_size: int,
+    shuffle: bool = False,
+    force_full_batches: bool = False,
+    device: Optional[str] = None,
+    tokenizer: Optional[Callable] = None,
+    max_len: Optional[int] = None,
+) -> np.ndarray:
+
+    backend = backend.lower()
+    model_kwargs = {
+        'model': model, 'batch_size': batch_size, 'tokenizer': tokenizer, 'max_len': max_len
+    }
+    if backend == 'tensorflow':
+        from alibi_detect.cd.tensorflow.preprocess import preprocess_drift
+    elif backend == 'pytorch':
+        from alibi_detect.cd.pytorch.preprocess import preprocess_drift  # type: ignore
+        model_kwargs['device'] = device
+    else:
+        raise NotImplementedError(f'{backend} not implemented. Use tensorflow or pytorch instead.')
+    model_fn = partial(preprocess_drift, **model_kwargs)
+
```

Comment:
If I understand correctly up until here the `get_preds` function only initialises the same `model_fn` function over and over for each function call. I therefore believe it would be better to move it up to initialization of the detector and pass it as input to `partial(classifier_uncertainty, model_fn=model_fn, ..., )`. This would avoid conditional importing and initialization for each batch at runtime and could make serialization a bit easier as well.